libs/market-list/src/lib/components/asset-details-dialog/asset-details-dialog.tsx

```diff
@@ -0,0 +1,163 @@
+import { gql, useQuery } from '@apollo/client';
+import { formatNumber, t, toBigNum } from '@vegaprotocol/react-helpers';
+import type { Asset } from '@vegaprotocol/react-helpers';
+import {
+  Button,
+  Dialog,
+  Icon,
+  Intent,
+  Splash,
+  Tooltip,
+} from '@vegaprotocol/ui-toolkit';
+import type {
+  AssetsConnection,
+  AssetsConnection_assetsConnection_edges_node_source_ERC20,
+} from './__generated__/AssetsConnection';
+
+export type AssetDetailsDialogState = {
+  isAssetDetailsDialogOpen: boolean;
+  assetDetailsDialogSymbol: string | Asset;
+};
+
+export const DEFAULT_ASSET_DETAILS_STATE: AssetDetailsDialogState = {
+  isAssetDetailsDialogOpen: false,
+  assetDetailsDialogSymbol: '',
+};
+
+type Dictionary<T> = {
+  [index: string]: T;
+};
+
+export const ASSETS_CONNECTION_QUERY = gql`
+  query AssetsConnection {
+    assetsConnection {
+      edges {
+        node {
+          id
+          name
+          symbol
+          totalSupply
+          decimals
+          quantum
+          source {
+            ... on ERC20 {
+              contractAddress
+              lifetimeLimit
+              withdrawThreshold
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export interface AssetDetailsDialogProps {
+  assetSymbol: string | Asset;
+  open: boolean;
+  onChange: (open: boolean) => void;
+}
+
+const ASSET_DETAILS_DICTIONARY: Dictionary<string> = {
+  Name: '', // 'Name of the asset (e.g: Great British Pound)',
+  Symbol: '', // 'Symbol of the asset (e.g: GBP)',
+  Decimals: 'Number of decimal / precision handled by this asset',
+  Quantum: 'The minimum economically meaningful amount in the asset',
+  TotalSupply: 'Total circulating supply for the asset',
+  ContractAddress:
+    'The address of the contract for the token, on the ethereum network',
+  WithdrawalThreshold:
+    'The maximum allowed per withdraw note: this is a temporary measure for restricted mainnet',
+  LifetimeLimit:
+    'The lifetime limits deposit per address note: this is a temporary measure for restricted mainnet',
+};
+
+const w = (key: string) =>
```

Comment:
If w is making labels from Dictionary keys, I would not go that way.
It will be much easier to maintain below structure
```
const details = [
 {
    value: (node: AssetsConnection_assetsConnection_edges_node) => node.name
    label: t('Name'),
    tooltip t('Name of the asset (e.g: Great British Pound)'),
  },
  ...
]
```

Objects is is an unordered collection of properties and Object.keys does not (always) follow the insertion order